movement/movement.h

```diff
@@ -312,4 +313,85 @@ void movement_play_alarm_beeps(uint8_t rounds, BuzzerNote alarm_note);
 
 uint8_t movement_claim_backup_register(void);
 
+/**
+ * The High Precision Timer (HPT) is an on-demand timer running at 1024hz.
+ * It runs independently of the real time clock and can be used to measure
+ * durations and schedule future tasks. The HPT is not affected by changes to 
+ * the RTC clock time. While enabled, it continues to run in standby mode and 
+ * may be used to wake the watch up from sleep.
+ * 
+ * The HPT is disabled by default to conserve power. Before using it to get a
+ * timestamp, a watch face must activate it using "movement_hpt_request". If
+ * not already running, this will enable and start the timer module. While the
+ * timer is enabled, the face may retrieve the current timestamp using
+ * "movement_hpt_get", or schedule a background event using
+ * "movement_hpt_schedule". When a face no longer needs to use the timestamp or
+ * scheduled event provided by the HPT it MUST call "movement_hpt_relenquish".
+ * If no other face has an outstanding request for the HPT, it will be disabled
+ * to conserve power.
+ * 
+ * Watch faces may not modify the value of the HPT counter in any way. The only
+ * guarantee to be made about the HPT timestamp is that between the time your
+ * face calls "movement_hpt_request" until the moment it calls
+ * "movement_hpt_relenquish", the value returned from "movement_hpt_get" will
+ * increment upwards at 1024hz. Outside of that window, the timestamp value may
+ * change unpredictably.
+ * 
+ * Faces may schedule an EVENT_HPT event to occur by calling 
+ * "movement_hpt_schedule" and passing in a timestamp for the event to occur.
+ * The face must call "movement_hpt_request" before scheduling the event, and
+ * must not call "movement_hpt_relenquish" until after the event has occurred.
```

Comment:
Why not? It won't be possible to cancel a running timer? What if the user cancels an alarm?

Reply:
Oh I mean, they can call "relenquish" if they want to, but it means their scheduled event is going to be cancelled, and some devs might be surprised by that.